### PR TITLE
Add wharton events

### PR DESCRIPTION
**Pull request by adamdom**
Functionality to scrape pages like:
https://spike.wharton.upenn.edu/calendar/index.cfm?format=list

It works for today or a given date. README updated as well. All times in EST.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -497,6 +497,57 @@ Given a date in <code>YYYY-MM-DD</code>format, returns all events occurring 2 we
...
+	</tr>
+	<tr>
+	    <td>Parameters</td>
+	    <td>None</td>
+	</tr>
+    </tbody>
+</table>
+
+### Get Events on Date
+Returns a list of events on a given date with the start time, end time, title, relevant link, and date. Date is given in format YYY-MM-DD.
```
Comment (esqu1) [inline README.md]: Correct date format

File: server/spikeCalendar.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      if len(times) == 2:
+        event["startTime"] = times[0]
+        event["endTime"] = times[1]
+        event["date"] = d.strftime('%Y-%m-%d')
+        events.append(event)
+  return jsonify({'calendar': events})
+
+
+@app.route('/whartonEvents/', methods=['GET'])
+def pull__wharton_event_today():
```
Comment (esqu1) [inline server/spikeCalendar.py]: Change two underscores to one

File: server/__init__.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
 import server.studyspaces
 import server.weather
 import server.calendar3year
+import server.spikeCalendar
```
Comment (adelq) [inline server/__init__.py]: No camel case modules

File: server/spikeCalendar.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      times = cleanSpikeTimeString(time_range)
+      if len(times) == 2:
+        event["startTime"] = times[0]
+        event["endTime"] = times[1]
+        event["date"] = d.strftime('%Y-%m-%d')
+        events.append(event)
+  return jsonify({'calendar': events})
+
+
+@app.route('/whartonEvents/', methods=['GET'])
```
Comment (adelq) [inline server/spikeCalendar.py]: Stop the camelCase - all lowercase

File: README.md [hunk 5]
```diff
@@ -497,6 +497,57 @@ Given a date in <code>YYYY-MM-DD</code>format, returns all events occurring 2 we
...
+## Wharton Calendar
+
+### Get Today's Events
+Returns a list of events for today with the start time, end time, title, relevant link, and date.
+
+<table>
+    <tbody>
+        <tr>
+	    <td>URL</td>
+	    <td><code>https://api.pennlabs.org/whartonEvents</td>
```
Comment (adelq) [inline README.md]: camelCase

File: server/spikeCalendar.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
+from server import app
+from bs4 import BeautifulSoup
+from flask import jsonify
+import re
+import requests
+import datetime
+
+BASE_URL = "https://spike.wharton.upenn.edu/calendar/index.cfm?format=list&date="
+
+def cleanSpikeTimeString(s):
```
Comment (adelq) [inline server/spikeCalendar.py]: No camelCase

File: server/spikeCalendar.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  regex = re.compile("1?[0-9]:[0-9][0-9] [AP]M")
+  all_times = regex.findall(s)
+  return all_times
```
Comment (adelq) [inline server/spikeCalendar.py]: All: use 4 spaces as is recommended in Python and for consistency with the rest of the project

File: server/spikeCalendar.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+def events_on_date(d):
+  """Pulls the calendar from the Penn website and
+  filters out which events are 2 weeks away from date d.
+
+  :param d: date object that specifies the date
+  """
+  r = requests.get(BASE_URL + d.strftime('%Y-%m/%d'))
```
Comment (adelq) [inline server/spikeCalendar.py]: Whitespace

File: server/spikeCalendar.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+def events_on_date(d):
+  """Pulls the calendar from the Penn website and
+  filters out which events are 2 weeks away from date d.
+
+  :param d: date object that specifies the date
+  """
+  r = requests.get(BASE_URL + d.strftime('%Y-%m/%d')) 
+  soup = BeautifulSoup(r.text, "html5lib")
+  cal = soup.find("body").find(id="content").find(id="calendar_content").find(id="calendar_event_list")
```
Comment (adelq) [inline server/spikeCalendar.py]: Does this need the first 2 `find`s? `id` is unique, so unless there's malformed HTML only the last one is needed

File: server/spikeCalendar.py [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+  return events_on_date(today)
+
+
+@app.route('/whartonEvents/<date>', methods=['GET'])
+def pull_wharton_event_on_date(date):
+  """Return JSON object with all events on the
+  date passed in as an argument.
+  """
+  d = datetime.datetime.strptime(date, "%Y-%m-%d").date()
+  return events_on_date(d)
```
Comment (adelq) [inline server/spikeCalendar.py]: Whitespace

File: server/spikeCalendar.py (hunk 6 above)
Comment (adelq) [inline server/spikeCalendar.py]: Also missing function docstring

File: server/spikeCalendar.py (hunk 9 above)
Comment (adamdom) [inline server/spikeCalendar.py]: It seemed to want me to work down the hierarchy. I tried just going straight for the id but, it didnt work. But I can try again.

**Comment by adamdom** [comment]
Just saw all these changes, will get to them soon.

**Pull request closed without merging**